server/cluster.go

```diff
@@ -483,14 +496,14 @@ func (c *Cluster) TopicMaster(msg *ClusterReq, rejected *bool) error {
 		sess, count = globals.sessionStore.NewSession(node, sid)
 
 		log.Println("cluster: topic proxy channel started", sid, count)
-		go sess.topicProxyWriteLoop()
+		go sess.topicProxyWriteLoop(msg.RcptTo)
 	}
 
 	switch {
 	case msg.TopicMsg.JoinReq != nil:
-		if msg.Sess.Uid > 0 {
-			log.Println("join setting uid = ", msg.Sess.Uid)
-			msg.CliMsg.from = msg.Sess.Uid.UserId()
+		if uid > 0 {
```

Comment:
@aforge 

I just noticed this. Let's use `uid.IsZero()` instead. For consistency and because `types.Uid` should be treated as opaque code. We may have to change it for federation. It would be a lot easier to change it if it's not assumed to be an `int`.

Reply:
Fixed in the followup PR.